File: picker/src/main/java/com/ds/magicpicker/ImageUtil.java
package com.ds.magicpicker;

import android.content.Context;
import android.content.DialogInterface;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;

/**
 * Created by deepak on 14/3/18.
 */

public class ImageUtil {
    public static void showImagePicker(final Context context, final int resultCode) {
        final CharSequence[] items = {context.getString(R.string.camera)
                , context.getString(R.string.gallery)};
        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder.setTitle(context.getString(R.string.add_photo));
        builder.setItems(items, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int item) {
                if (items[item].equals(context.getString(R.string.camera))) {
                    ((AppCompatActivity) context).startActivityForResult(new PickerBuilder(context).openCamera(true).build(), resultCode);
                } else if (items[item].equals(context.getString(R.string.gallery))) {
                    ((AppCompatActivity) context).startActivityForResult(new PickerBuilder(context).openGallery(true).build(), resultCode);
                }
            }
        });
        builder.show();
    }
}

File: picker/src/main/java/com/ds/magicpicker/ImageUtil.kt
@file:JvmName("ImageUtils")
package com.ds.magicpicker

import android.content.Context
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity

/**
 * Created by deepaksingh on 25/05/19.
 */
var CONST_IMAGE_URI = "image_uri"
var CONST_OPTION_CAMERA = "open_camera"
var CONST_OPTION_GALLERY = "open_gallery"
val CONST_PICK_IMAGE = 103

fun showImagePicker(context: Context, resultCode: Int) {
    val items = arrayOf<CharSequence>(context.getString(R.string.camera), context.getString(R.string.gallery))
    val builder = AlertDialog.Builder(context)
    builder.setTitle(context.getString(R.string.add_photo))
    builder.setItems(items) { dialog, item ->
        if (items[item] == context.getString(R.string.camera)) {
            (context as AppCompatActivity).startActivityForResult(PickerBuilder(context).openCamera().build(), resultCode)
        } else if (items[item] == context.getString(R.string.gallery)) {
            (context as AppCompatActivity).startActivityForResult(PickerBuilder(context).openGallery().build(), resultCode)
        }
    }
    builder.show()
}